Refactor installation documentation

geriom opened the pull request:
# Changes

- Simplify installation instructions
- Move configuration docs to a separate file.
- Include vendor specific docs for OpenShift and Google Cloud.

As discussed in the proposal for this change during the community meeting, the documentation is only rendered correctly on the website. This and the [corresponding PR on the website repo](https://github.com/tektoncd/website/pull/482) must be merged in short succession.

## Previews

- [Pipelines installation](https://deploy-preview-469--tekton.netlify.app/docs/installation/pipelines/)
- [Post-installation configurations](https://deploy-preview-469--tekton.netlify.app/docs/installation/post-install/)
- [Registry authentication](https://deploy-preview-469--tekton.netlify.app/docs/how-to-guides/kaniko-build-push/#container-registry-authentication)

/kind documentation

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:


- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

bendory commented:
Forgive my ignorance if this is a newbie question... is there a staging area where I can see the rendered website as it will appear after the merge?

lbernick commented on docs/install.md:
```diff
@@ -1,717 +1,132 @@
+## Additional configuration options
 
-## Verify Tekton Pipelines Release
+You can enable additional alpha an beta features, customize execution
```
```suggestion
You can enable additional alpha and beta features, customize execution
```

lbernick commented on docs/install.md:
```diff
@@ -1,717 +1,132 @@
-If you want to run OpenShift 4.x on your laptop (or desktop), you
-should take a look at [Red Hat CodeReady Containers](https://github.com/code-ready/crc).
-
-## Configuring PipelineResource storage
```
It looks like this block is being removed, but I'm not sure whether we should remove PipelineResources docs yet even though they are deprecated. Maybe we could just move to a less prominent location?

geriom replied:
I moved that Openshift note to the Openshift tab, you can find it at the bottom of that tab here: https://deploy-preview-469--tekton.netlify.app/docs/installation/pipelines/

As for the PipelineResources storage section, I just added it back in the post-installation configurations.

Thank you!

lbernick reviewed (approved):
Thanks Geri! I really like the tabs for installing on different providers/with different registries!

geriom commented:
> Forgive my ignorance if this is a newbie question... is there a staging area where I can see the rendered website as it will appear after the merge?

The links in the "Preview" section take you there. For changes outside the website repo the preview is not autogenerated, I created a separate PR just to generate the preview.

geriom commented:
/retest

jerop reviewed (approved):
/lgtm

jerop commented:
/retest

afrittoli commented on docs/install.md:
```diff
@@ -1,809 +1,131 @@
+## Prerequisites
 
-   Or, for the nightly release, use:
+-   A [Kubernetes cluster][k8s] running version 1.22 or later.
```
This should be 1.23 or later

geriom replied:
Updated.

afrittoli commented on docs/install.md:
```diff
@@ -1,809 +1,131 @@
+your computer.
 
-   You can install a specific release using `previous/$VERSION_NUMBER`. For example:
+[k8s]: https://kubernetes.io/docs/concepts/overview/what-is-kubernetes/
+[kubectl]: https://kubernetes.io/docs/tasks/tools/#kubectl
+[rbac]: https://kubernetes.io/docs/reference/access-authn-authz/rbac/
+[metrics]: https://github.com/kubernetes-sigs/metrics-server
+[local-install]: https://tekton.dev/docs/installation/local-installation/
```
NIT: perhaps we could have all the links at the bottom of the doc

geriom replied:
I moved them to the bottom.

afrittoli reviewed (commented):
Thanks for this @geriom !
A couple of questions:
- we have no way to see here the final rendering result of this PR. Would it make sense to have a PR against the website repo that pulls in this PR and shows the final result?
- I like that we have tabs with different installation options. Should we have one about using the operator that points to the operator docs as well?
- NIT: I'm not entirely sure about the "post-install" naming. The configuration is not exclusively done *after* the installation, the installation files can also be customised to install directly with the required configuration

geriom commented:
> * we have no way to see here the final rendering result of this PR. Would it make sense to have a PR against the website repo that pulls in this PR and shows the final result?

I was thinking the same thing. For this PR I rendered it locally [see preview here](https://deploy-preview-469--tekton.netlify.app/docs/installation/pipelines/), but that's not a feasible solution long term. Not sure how much engineering that would require, but it's probably worth the effort. I opened an [issue on the website repo](https://github.com/tektoncd/website/issues/493) to track this

> * I like that we have tabs with different installation options. Should we have one about using the operator that points to the operator docs as well?

Yes, I think that's a good idea. We can make that a separate issue and iterate this doc? since this PR is been open for two moths now.

> * NIT: I'm not entirely sure about the "post-install" naming. The configuration is not exclusively done _after_ the installation, the installation files can also be customized to install directly with the required configuration.

Renamed the file to `additional-configs.md` and changed the title to "Additional configuration options", what do you think?

afrittoli commented:
Thanks for the updates!
/lgtm
